components/Modals/AddItemRequirementModal.tsx

```diff
@@ -0,0 +1,247 @@
+import {
+  Button,
+  Flex,
+  Image,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  useRadioGroup,
+  useToast,
+  VStack,
+} from '@chakra-ui/react';
+import { useCallback, useEffect, useState } from 'react';
+import { parseAbi } from 'viem';
+import { Address, usePublicClient, useWalletClient } from 'wagmi';
+
+import { RadioCard } from '@/components/RadioCard';
+import { TransactionPending } from '@/components/TransactionPending';
+import { useGame } from '@/contexts/GameContext';
+import { useItemActions } from '@/contexts/ItemActionsContext';
+import { waitUntilBlock } from '@/hooks/useGraphHealth';
+
+export const AddItemRequirementModal: React.FC = () => {
+  const { character, game, reload: reloadGame } = useGame();
+  const { selectedItem, addRequirementModal } = useItemActions();
+
+  const { data: walletClient } = useWalletClient();
+  const publicClient = usePublicClient();
+  const toast = useToast();
+
+  const [classId, setClassId] = useState<string>('1');
+
+  const [isAdding, setIsAdding] = useState<boolean>(false);
+  const [txHash, setTxHash] = useState<string | null>(null);
+  const [isSyncing, setIsSyncing] = useState<boolean>(false);
+  const [isSynced, setIsSynced] = useState<boolean>(false);
+
+  // TODO: For now we are only adding class requirements
+  // When the new contracts are deployed, we will add item requirements too
+  const options = game?.classes.map(c => c.classId) ?? [];
```

Comment:
Here we are display all classes. We want to display only options that are NOT already requirements, right?

Reply:
Oh, yeah definitely... I actually also think this is using a function call from the old contracts, so it may not even work atm

Reply:
Will address this in a separate PR